common/src/main/java/com/microsoft/identity/common/internal/eststelemetry/SharedPreferencesLastRequestTelemetryCache.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.eststelemetry;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.common.internal.cache.ISharedPreferencesFileManager;
+import com.microsoft.identity.common.internal.logging.Logger;
+
+import java.util.Map;
+
+public class SharedPreferencesLastRequestTelemetryCache extends SharedPreferencesRequestTelemetryCache {
```

Comment:
Unless `save()` and `get()` are in the same class, what is the purpose of a subclass? Isn't it important to know how `save()` works to write the `get()` function?

How would a second subclass implement new functionality without modifying the parent class? Wouldn't they have to `@Override` the `save()` method? If yes, why not declare both `save()` and `get()` as `abstract` in the `super` class?

Reply:
@iambmelt You're right, this is messed up.

I think I'll just remove the `SharedPreferencesRequestTelemetryCache` class, and instead just implement the methods directly in the `SharedPreferencesLastRequestTelemetryCache` class. I don't think we are ever going to be tracking any request beyond the last request so I think just having one class will be simpler.

Reply:
Addressed as stated above.